zadaci/sa-kolokvijuma/2020/E1/T34/G8/resenje.c

```diff
@@ -0,0 +1,123 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define MAX_PREVOZNIK 30+1
+#define MAX_STANJE 10+1
+#define MAX_NIZ 10
+
+typedef struct autobus_st
+{
+    char prevoznik[MAX_PREVOZNIK];
+    char stanje[MAX_STANJE];
+    double cena_karte;
+    unsigned broj_mesta;
+}Autobus;
+
+FILE *otvori(char *name,char *mode)
+{
+    FILE *f = fopen(name,mode);
+
+    if(f == NULL)
+    {
+        printf("Problem prilikom otvaranja %s datoteke.",name);
+        exit(-2);
+    }
+
+    return f;
+}
+
+void ucitavanje(FILE *in, Autobus *a, int *n)
+{
+    *n=0;
+    while(fscanf(in,"%s %s %lf %u",
+                    a[*n].prevoznik,
+                    a[*n].stanje,
+                    &a[*n].cena_karte,
+                    &a[*n].broj_mesta) != EOF)
+    {
+        (*n)++;
+    }
+}
+
+void letovanje(FILE *out,Autobus *a, int n, int prijav_put,char *tr_prev)
+{
+    int i, suma=0;
+    for(i=0;i<n;i++)
+    {
+        if(strcmp(a[i].prevoznik,tr_prev) == 0 
+            && strcmp(a[i].stanje,"ispravan") == 0)
+        {
+            suma+=a[i].broj_mesta;
+        }
+    }
+
+    if(suma >= prijav_put)
+    {
+        fprintf(out,"\nPrevoznik %s na letovanje moze da preveze",tr_prev);
+        fprintf(out," sve prijavljene putnike.");
```

Comment:
Just so you know, you could have split the string in multiple lines. My favorite way of doing it in C is:

```
fprintf(out,
          "\nPrevoznik %s na letovanje moze da preveze"
          " sve prijavljene putnike.",
          tr_prev
);
```

You don't have to correct this.